File: b_rank_skillcheck_archive/max_average_segment_05.py
# 実行時間 130ms
n, k = list(map(int, input().split()))
a = list(map(int, input().split()))
b = [0] * (n - k + 1)
mc = 1
mi = -1
mx = -1
s = sum(a[:k])
l = a[k:]

for i in range(len(l)):
    l[i] -= a[i]
    b[i + 1] = b[i] + l[i]
for i in range(len(b)):
    b[i] += s
    if mx < b[i]:
        mc = 1
        mi = i + 1
        mx = b[i]
    elif mx == b[i]:
        mc += 1
print(mc, mi)

File: b_rank_skillcheck_archive/max_average_segment_05.cs
// 実行時間 110ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int[] n = ReadLine().Split().Select( int.Parse ).ToArray();
        int[] a = ReadLine().Split().Select( int.Parse ).ToArray();
        int[] b = new int[n[0] - n[1] + 1];
        int mc = 1;
        int mi = -1;
        int mx = -1;
        int s = a.Take( n[1] ).Sum();
        var l = new System.Collections.Generic.List<int>( a.Skip( n[1] ) );

        foreach( int i in Range( 0, l.Count ) ) {
            l[i] -= a[i];
            b[i + 1] = b[i] + l[i];
        }
        foreach( int i in Range( 0, b.Length ) ) {
            b[i] += s;
            if( mx < b[i] ) {
                mc = 1;
                mi = i + 1;
                mx = b[i];
            } else if( mx == b[i] ) {
                mc++;
            }
        }
        WriteLine( $"{mc} {mi}" );
        return;
    }
}
